collect_app/src/main/java/org/odk/collect/android/preferences/PreferencesRepository.kt

```diff
@@ -0,0 +1,48 @@
+package org.odk.collect.android.preferences
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+
+class PreferencesRepository(private val context: Context) {
+    @JvmOverloads
+    fun getMetaPreferences(projectId: String = ""): PreferencesDataSource {
+        val preferenceId = META_PREFS_NAME + projectId
```

Comment:
Actually thinking about this again I'm not sure this is the behaviour we'll want. For background jobs we'll need to pass the project ID but in most places we'll just want the no arg case to return the "current" project. As far as I can see the parameter is never passed in, so maybe we should just leave the projectId stuff out until we're actually switching between projects.

Reply:
`getMetaPreferences()` is not the best example because in fact it should be unique and not to accept any `projectID`, I fixed it.
When it comes to general and admin preferences yes for background jobs that parameter will be used otherwise we probably should take current project id from meta preferences.

I'll add tests for this class but maybe a bit later one we decide for 100% what we need.